library/private/options_tools.h

```diff
@@ -428,6 +448,17 @@ std::string format(direction_t var)
   }
 }
 
+//----------------------------------------------------------------------------
+/**
+ * Format provided var into a string from provided colormap_t.
+ * Rely on `format(std::vector<double>&)`
+ * TODO add proper formatting
+ */
+std::string format(colormap_t var)
```

Comment:
Any reason it's passed by copy?
```suggestion
std::string format(const colormap_t& var)
```

Reply:
no reason ^^